src/plugin_discovery/plugin_exports/__tests__/reduce_export_specs.js

```diff
@@ -0,0 +1,77 @@
+import expect from 'expect.js';
+
+import { PluginPack } from '../../plugin_pack';
+import { reduceExportSpecs } from '../reduce_export_specs';
+
+const PLUGIN = new PluginPack({
+  path: __dirname,
+  pkg: {
+    name: 'foo',
+    version: 'kibana'
+  },
+  provider: ({ Plugin }) => (
+    new Plugin({
+      uiExports: {
+        concatNames: {
+          name: 'export1'
+        },
+
+        concat: [
+          'export2',
+          'export3',
+        ],
+      }
+    })
+  )
+});
+
+const REDUCERS = {
+  concatNames(acc, spec, type, pluginSpec) {
+    return {
+      names: [].concat(
+        acc.names || [],
+        `${pluginSpec.getId()}:${spec.name}`,
```

Comment:
do we actually need the entire `pluginSpec` passed in to the reducers?

Reply:
I assume no, but by doing so `reduceExportSpecs()` doesn't need to know what is actually needed by the reducers and gives them room to grow.